rust/scx_utils/src/topology.rs

```diff
@@ -692,10 +711,100 @@ fn is_smt_active() -> Option<bool> {
+            let core_type = core.core_type.clone();
+            cores_by_type
+                .entry(core_type)
+                .or_insert(Vec::new())
+                .push(*core_id);
+        }
+
+        for (_core_type, core_ids) in cores_by_type.iter() {
+            let num_cores_in_bucket = core_ids.len();
+
+            let best_split = find_best_split_size(num_cores_in_bucket, min_cores, max_cores);
+            let num_partitions = num_cores_in_bucket / best_split;
+
+            for (bucket_idx, &core_id) in core_ids.iter().enumerate() {
+                let partition_idx = min(bucket_idx / best_split, num_partitions - 1);
+                let current_partition_id = partition_id + partition_idx;
+                core_to_partition.insert(core_id, current_partition_id);
+                partition_to_kernel_id.insert(current_partition_id, llc.kernel_id);
+            }
+
+            partition_id += num_partitions;
+            total_partitions = partition_id;
+        }
+    }
+
+    let mut virt_llcs: BTreeMap<usize, Arc<Llc>> = BTreeMap::new();
+
+    for partition_id in 0..total_partitions {
+        let kernel_id = partition_to_kernel_id
+            .get(&partition_id)
+            .copied()
+            .unwrap_or(partition_id);
+        virt_llcs.insert(
+            partition_id,
+            Arc::new(Llc {
+                id: partition_id,
+                kernel_id,
+                cores: BTreeMap::new(),
+                span: Cpumask::new(),
+                node_id: node.id,
+                all_cpus: BTreeMap::new(),
+            }),
+        );
+    }
+
+    // Second pass: move cores to the appropriate new LLC based on partition
+    for (_llc_id, llc) in node.llcs.iter_mut() {
+        for (core_id, core) in llc.cores.iter() {
+            if let Some(&target_partition_id) = core_to_partition.get(core_id) {
+                if let Some(target_llc) = virt_llcs.get_mut(&target_partition_id) {
+                    let target_llc_mut = Arc::get_mut(target_llc).unwrap();
+
+                    let mut new_core = (**core).clone();
+                    new_core.llc_id = target_partition_id;
+
+                    let mut updated_cpus = BTreeMap::new();
+                    for (cpu_id, cpu) in new_core.cpus.iter() {
+                        let mut new_cpu = (**cpu).clone();
+                        new_cpu.llc_id = target_partition_id;
+
+                        target_llc_mut.span.set_cpu(*cpu_id)?;
+
+                        updated_cpus.insert(*cpu_id, Arc::new(new_cpu));
+                    }
+                    new_core.cpus = updated_cpus;
+
+                    target_llc_mut.cores.insert(*core_id, Arc::new(new_core));
+                }
+            }
+        }
+    }
+
+    node.llcs = virt_llcs;
+
+    Ok(())
+}
+
 fn create_default_node(
     online_mask: &Cpumask,
     topo_ctx: &mut TopoCtx,
     flatten_llc: bool,
+    min_cores: Option<usize>,
+    max_cores: Option<usize>,
```

Comment:
Given that these params are meaningful only when both are specified: sth like `nr_cores_per_vllc: Option<(usize, usize)>`?